### PR TITLE
move repos package from repo-updater into internal/

### PR DESCRIPTION
**Pull request by arussellsaw**
There are a bunch of places outside of repo-updater that use the repos package, and with the work @tsenart has proposed to deprecate repo-updater (https://github.com/sourcegraph/sourcegraph/issues/16197) it makes more sense for this code to live outside of repo-updater.

This is directly a precursor to my work to add a new graphql endpoint to list affiliated repositories for an external service: https://github.com/sourcegraph/sourcegraph/issues/16501

apologies for the PR size, but there should be no actual code changes.

### CODE REVIEW

File: internal/repos/github.go [hunk 1]
```diff
@@ -776,3 +777,35 @@ func exampleRepositoryQuerySplit(q string) string {
 	_ = enc.Encode(qs)
 	return strings.TrimSpace(b.String())
 }
+
+func (s *GithubSource) AffiliatedRepositories(ctx context.Context, opts AffiliatedRepositoryOptions) ([]types.AffiliatedRepository, error) {
```
Comment (tsenart) [inline internal/repos/github.go:781]: I think you meant to leave this addition to a later PR?

File: internal/types/types.go [hunk 2]
```diff
@@ -396,6 +396,12 @@ func (rs Repos) Filter(pred func(*Repo) bool) (fs Repos) {
 	return fs
 }
 
+type AffiliatedRepository struct {
```
Comment (tsenart) [inline internal/types/types.go:399]: Also this.

File: monitoring/monitoring/README.md [hunk 3]
```diff
@@ -36,15 +36,15 @@ To learn more about the generator\, see the top\-level program: https://github.c
 - [type UnitType](<#type-unittype>)
 
 
-## func [Generate](<https://github.com/sourcegraph/sourcegraph/blob/main/monitoring/monitoring/generator.go#L25>)
```
Comment (tsenart) [inline monitoring/monitoring/README.md:39]: Why has this file changed?

**Review by tsenart — APPROVED** [review]

**Review by eseliger — APPROVED** [review]
campaigns changes lgtm

File: internal/repos/github.go (hunk 1 above)
Comment (ryanslade) [inline internal/repos/github.go:781]: I don't think these changes should be included

File: internal/repos/sources.go [hunk 4]
```diff
@@ -126,6 +126,15 @@ type ChangesetSource interface {
 	ReopenChangeset(context.Context, *Changeset) error
 }
 
+type AffiliatedRepositorySource interface {
```
Comment (ryanslade) [inline internal/repos/sources.go:129]: I don't think these changes should be included

File: internal/types/types.go (hunk 2 above)
Comment (ryanslade) [inline internal/types/types.go:399]: I don't think these changes should be included

File: monitoring/monitoring/README.md (hunk 3 above)
Comment (ryanslade) [inline monitoring/monitoring/README.md:39]: Looks like the generator isn't able to produce links now

**Review by ryanslade — APPROVED** [review]

File: monitoring/monitoring/README.md (hunk 3 above)
Comment (arussellsaw) [reply monitoring/monitoring/README.md:39]: i'm not sure 🤔

File: internal/repos/github.go (hunk 1 above)
Comment (arussellsaw) [reply internal/repos/github.go:781]: ah yep good catch

File: monitoring/monitoring/README.md (hunk 3 above)
Comment (arussellsaw) [reply monitoring/monitoring/README.md]: resolved, seems like it's something wrong with a local version of some package on my end

**Pull request merged**